sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -289,6 +289,9 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]] extends TreeNode[PlanT
       case m: Map[_, _] => m
       case d: DataType => d // Avoid unpacking Structs
       case seq: Traversable[_] => seq.map(recursiveTransform)
+      case ExistenceJoin(exists: Attribute) =>
+        // `exists` must be an Attribute in ExistenceJoin
+        ExistenceJoin(transformExpression(exists).asInstanceOf[Attribute])
```

Comment:
Why is this change necessary? This smells for two reasons:

- `QueryPlan` shouldn't have know about such a detail. This means something is wrong with the abstraction.
- Why do we want to change the attribute in an ExistenceJoin? It is produced by the join, and we really should not replace it.